x-pack/plugins/maps/public/classes/sources/es_agg_source/es_agg_source.ts

```diff
@@ -33,9 +33,21 @@ export abstract class AbstractESAggSource extends AbstractESSource {
   private readonly _metricFields: IESAggField[];
   private readonly _canReadFromGeoJson: boolean;
 
+  static createDescriptor(
+    descriptor: Partial<AbstractESAggSourceDescriptor>
+  ): AbstractESAggSourceDescriptor {
+    descriptor = AbstractESSource.createDescriptor(descriptor);
+    return {
+      ...descriptor,
+      indexPatternId: descriptor.indexPatternId ? descriptor.indexPatternId : '',
```

Comment:
createDescriptor should through if indexPatternId and type are not provided. Better to fail early.